components/fxa-client/android/src/main/java/mozilla/appservices/fxaclient/FirefoxAccount.kt

```diff
@@ -302,17 +302,33 @@ class FirefoxAccount(handle: FxaHandle, persistCallback: PersistCallback?) : Aut
      * @param scopes Array of scopes for the OAuth code.
      * @param state OAuth flow state.
      * @param accessType Type of access, "offline" or "online".
+     * @param codeChallenge Code challenge for Proof Key for Code Exchange (PKCE)
+     * @param codeChallengeMethod for PKCE, currently only 'S256' supported
+     * @param keysJwk Key to encrypt keys_jwe using, passed in as a base64 string
      * This performs network requests, and should not be used on the main thread.
      */
     fun authorizeOAuthCode(
         clientId: String,
         scopes: Array<String>,
         state: String,
-        accessType: String = "online"
+        accessType: String = "online",
+        codeChallenge: String,
+        codeChallengeMethod: String = "S256",
+        keysJwk: String
```

Comment:
/cc @vladikoff, this might be another good candidate for accepting an `OAuthParams` struct of some sort, rather than a bunch of individual parameters.

(For context: in our 1:1 earlier today @vladikoff and I were discussing how we don't really have a good example of how to pass a rich structure like that *in* to the Rust code; we mostly use the protobuf message types for passing things *out* of Rust).